plugins/xConf/xConf.pl

```diff
@@ -38,102 +38,121 @@ package xConf;
 );
 
 sub Unload {
-	Commands::unregister($chooks);
+	Commands::unregister($plugin_commands);
 	message "xConf plugin reloading or unloading\n", 'success'
 }
 
 sub xConf {
-my ($cmd, $args) = @_;
-	my ($file,$file2,$found,$key,$oldval,$shopname,$type,$value, $name, $inf_hash, $ctrl_hash);
+	my ($cmd, $args) = @_;
+	my ($file,$tables_file,$found,$key,$oldval,$shopname,$type,$value, $name, $inf_hash, $ctrl_hash, $id, $needQuotes);
+	
+	($key, $value) = $args =~ /([\s\S]+?)\s([\-\d\.]+[\s\S]*)/ if !$shopname;
+	
 	if ($cmd eq 'sconf') {
-		($key, $value) = $args =~ /(name)(?:\s)(.*)/;
-		$shopname = 1 if $key eq 'name';
-	}
-	($key, $value) = $args =~ /([\s\S]+?)(?:\s)([\-\d\.]+[\s\S]*)/ if !$shopname;
-	$key = $args if !$key;
-	$key =~ s/^\s+|\s+$//g;
-	debug "extracted from args: KEY: $key, VALUE: $value\n";
-	if (!$key) {
-		error "Syntax Error in function '$cmd'. Not found <key>\n".
-				"Usage: $cmd <key> [<value>]\n";
-		return;
-	}
-	if ($cmd eq 'iconf') {
+		($key, $value) = $args =~ /(name)\s(.*)/;
+		$shopname = $value if $key eq 'name';
+		$inf_hash = \%items_lut;
+		$ctrl_hash = \%shop;
+		$file = 'shop.txt';
+		$tables_file = 'tables\..\items.txt';
+		$type = 'Item';
+		
+	} elsif ($cmd eq 'iconf') {
+		
 		$inf_hash  = \%items_lut;
 		$ctrl_hash = \%items_control;
 		$file = 'items_control.txt';
-		$file2 = 'tables\..\items.txt';
+		$tables_file = 'tables\..\items.txt';
 		$type = 'Item';
-	} elsif ($cmd eq 'mconf') {
-		$inf_hash  = \%monsters_lut;
-		$ctrl_hash = \%mon_control;
-		$file = 'mon_control.txt';
-		$file2 = 'tables\..\monsters.txt';
-		$type = 'Monster';
+		
+		#syntax checking for items with number in name buy whithout quotes to surrond it
+		if ($key =~ /[a-zA-Z][0-9]/ && $key !~ /"/) {
```

Comment:
Doesn't cover latin or chinese caracters, for example

Reply:
solved